packages/contentful/src/delivery-api/poll-delivery-api.ts

```diff
@@ -0,0 +1,26 @@
+import { Entry as CDAEntry, EntrySkeletonType, FieldsType } from 'contentful';
+import retry from 'async-retry';
+
+export const pollContentfulDeliveryApi = async <
```

Comment:
to be fair the two functions are so similar i would be tempted to write one generic function to fetch and bail and then have these two higher order functions reuse that other one, just pass the right data extraction logic. Basically introduce some reusability maybe? 

Also it feels like both could just be part of one util file